### PR TITLE
Rails Blog App: Add forms

### PR DESCRIPTION
**Pull request by kkolade**
# In this PR, I:

- Create a method called current_user in the ApplicationController to make current user data available to all controllers.
- Create forms to perform the following functions:
   - Creates a Post on behalf of the current_user (using the method I created in the ApplicationController).
   - Create a comment on behalf of the current_user (using the method created in the ApplicationController).
- Allow Users to add likes to Posts.
- Add minimal styling to the forms and buttons created

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
+      <div class="comment-form">
+        <%= form_with model: [@user, @post, @comment], url: user_post_comments_path(@user, @post), class: "form", local: true do |form| %>
+          <div>
+            <%= form.text_area :text, class: "text-area", placeholder: "Add a comment..." %>
+          </div>
+          <div >
+            <%= form.submit "Add Comment", class: "button comment-button" %>
+          </div>
+        <% end %>
+      </div>
```
Comment (laiifuu) [inline app/views/posts/show.html.erb:39]: Good work adding a form for the user to add a comment :+1: However this form is supposed to be in a the `views/comments/new.html.erb` file.

File: app/controllers/likes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  def create
+    @user = current_user
+    @post = @user.posts.find(params[:post_id])
+    @like = Like.new(user_id: current_user.id, post: @post)
+
+    if @like.save
+      redirect_to user_post_path(params[:user_id], params[:post_id]), notice: 'like successfully saved.'
+    else
+      redirect_to user_post_path(params[:user_id], params[:post_id]), alert: 'like could not be saved.'
+    end
```
Comment (laiifuu) [inline app/controllers/likes_controller.rb]: After creating a post, I tried adding a like, and it didn't work, the likes counter didn't update. I accessed the list of likes from the rails consoleto make sure, and it was empty: 
image
Kindly fix this issue.

**Review by laiifuu — CHANGES_REQUESTED** [review]
## Hi @kkolade :wave: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close


## Highlights
- [x] Green linters check :heavy_check_mark: 
- [x] Descriptive PR summary :heavy_check_mark: 

## Required Changes ♻️
- [ ] When I tried accessing the page for a single user when that user had no posts, I got the following error: 
image
Kindly fix this issue.

_Check the comments under the review. :point_down:_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kkolade** [comment]
Hi @laiifuu please are you available for a chat on slack

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @kkolade, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request :red_circle: 

You have submitted a project, that was not approved in the previous week. I would like to suggest that you close this PR `without` merging it, then create a new PR and use the new PR to request a review again.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request closed without merging**